### PR TITLE
added authentication for all comment routes

### PR DESCRIPTION
**Pull request by rusye**
#80 #79 #78 #77 #76

### CODE REVIEW

File: src/routes/comments.js [hunk 1]
```diff
@@ -84,34 +109,54 @@ Create mongoose ObjectId from parameter
...
+    .then(comment => {
+      if (!comment) {
+        return res.status(404).json({ Error: 'Comment not found' });
+      }
+
+      if (req.user._id !== comment.user.toString()) {
+        return res.status(403).json({ message: "This isn't your comment" });
       }
+
+      comment.remove().then(comment => res.status(204).end());
```
Comment (a-w-m) [inline src/routes/comments.js]: Any reason to use `findById()` and `remove()` over `findByIdAndDelete()`? Just curious

File: src/routes/comments.js [hunk 2]
```diff
@@ -84,34 +109,54 @@ Create mongoose ObjectId from parameter
...
+  const { text: textToUpdate } = req.body;
+  const { name } = req.user;
+
+  Comment.findById(req.params.commentID)
+    .then(comment => {
+      if (!comment) {
+        return res.status(404).json({ Error: 'Comment not found' });
+      }
+
+      if (req.user._id !== comment.user.toString()) {
```
Comment (a-w-m) [inline src/routes/comments.js:141]: Oh okay, it seems this way allows you to verify correct user_id coming in from request. Great.

File: src/routes/comments.js (hunk 1 above)
Comment (rusye) [reply src/routes/comments.js]: @a-w-m `findByIdAndDelete()` will immediately delete the comment, I wanted to first `findById()` and then check that the `comment.user === jwtAuth.user._id` before allowing it to `.remove()`.

File: src/routes/comments.js (hunk 2 above)
Comment (rusye) [reply src/routes/comments.js:141]: @a-w-m ah you have answered you're own question lol, I was answering questions as I scrolled down

File: src/routes/comments.js [hunk 3]
```diff
@@ -84,34 +106,54 @@ Create mongoose ObjectId from parameter
...
-
-  Comment.findByIdAndUpdate(
-    new mongoose.Types.ObjectId(req.params.commentID),
-    { text },
-    { upsert: false, new: true }
-  )
-    .then(comment => res.json(comment))
-    .catch(err => res.status(404).json({ error: "Something went wrong" }));
+router.patch('/:commentID', (req, res) => {
+  const { text: textToUpdate } = req.body;
```
Comment (Junjpan) [inline src/routes/comments.js:132]: I don't understand why you did 'const {text:textToUpdate}=req.body' instead of 'const {text}=req.body'

**Review by Junjpan — APPROVED** [review]

**Review by a-w-m — APPROVED** [review]

**Pull request merged**